ebpf-based/Makefile

```diff
@@ -0,0 +1,21 @@
+# Obtain an absolute path to the directory of the Makefile.
+# Assume the Makefile is in the root of the repository.
+REPODIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+
+# Build the list of include directories to compile the bpf program
+BPF_INCLUDE += -I${REPODIR}/include/libbpf
+BPF_INCLUDE+= -I${REPODIR}/include
+
+.PHONY: generate
+generate: export CFLAGS := $(BPF_INCLUDE)
+generate:
+	go mod tidy
+	go generate ./...
+
+.PHONY: build
```

Comment:
When i try to build the code , am getting an error 
```
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o kv-go-instrumentation cli/main.go
# github.com/keyval-dev/opentelemetry-go-instrumentation/pkg/opentelemetry
pkg/opentelemetry/controller.go:119:24: undefined: otlptracegrpc
pkg/opentelemetry/controller.go:120:3: undefined: otlptracegrpc
make: *** [Makefile:17: build] Error 2
```
Can we also make a note that we need to build this on linux (it might be obvious few folks , but initially i tried it on mac which failed with a different error ). 

Also can we rename the binary from ` kv-go-instrumentation` to ` opentelemetry-go-instrumentation`

Reply:
You can build it on Mac/Windows using the provided Dockerfile by running:
```
make docker-build IMG=otel-go-auto-inst:v0.1
```
I will document it better and change the binary name.